MapboxCoreNavigation/RouteController.swift

```diff
@@ -651,6 +662,27 @@ extension RouteController: CLLocationManagerDelegate {
         checkForFasterRoute(from: location)
     }
     
+    func detectRouteStepInTunnel(for location: CLLocation) {
+        guard routeProgress.currentLegProgress.currentStep.containsTunnel else { return }
+        guard let tunnelDistance = routeProgress.currentLegProgress.currentStep.tunnelDistance, let tunnelSlice = routeProgress.currentLegProgress.currentStep.tunnelSlice else { return }
+        guard let tunnelStartCoordinate = tunnelSlice.coordinates.first, let tunnelEndCoordinate = tunnelSlice.coordinates.last else { return }
+        
+        // Calculated distances from current location to tunnel entrance and exit
+        let currentLocation = CLLocationCoordinate2D(latitude: location.coordinate.latitude, longitude: location.coordinate.longitude)
+        let distanceToEntrance = currentLocation.distance(to: tunnelStartCoordinate)
+        let distanceToExit = currentLocation.distance(to: tunnelEndCoordinate)
+        
+        // Capute any location which lies within the tunnel route.
+        // Pre-conditions:
+        //   - Current location to exit must be greater than zero meters.
+        //   - Distance to tunnel entrance must be less than tunnel distance.
+        if (tunnelDistance - distanceToExit) > 0 && distanceToEntrance <= tunnelDistance {
```

Comment:
Would something like 

```swift
if distanceToEntrance <= tunnelDistance && distanceToExit <= tunnelDistance {
...
```

I only suggest this since it's easier to understand what we're comparing on each side of the && operator.